launcher/helper.cpp

```diff
@@ -89,8 +89,36 @@ QString getRealPath(QString path)
 void performNativeCopy(QString src, QString dst)
 {
 #ifdef VCMI_ANDROID
-	auto srcStr = QAndroidJniObject::fromString(src);
-	auto dstStr = QAndroidJniObject::fromString(dst);
+	auto percentEncodeNonAscii = [](const QString &input) {
+		QByteArray utf8 = input.toUtf8();
+		QByteArray encoded;
+
+		for (char c : utf8)
+		{
+			// If ASCII (0x00 to 0x7F), keep as is
+			if (static_cast<unsigned char>(c) < 0x80)
+				encoded.append(c);
+			else
+			{
+				// Non-ASCII: encode as %HH
+				encoded.append('%');
+				encoded.append(QByteArray::number(static_cast<unsigned char>(c), 16).toUpper().rightJustified(2, '0'));
```

Comment:
there's no overload with `unsigned char`, better use `uint`
```suggestion
				encoded.append(QByteArray::number(static_cast<uint>(c), 16).toUpper().rightJustified(2, '0'));
```